File: app/src/main/java/fr/piotr/dismoitoutsms/util/WakeLockManager.java
package fr.piotr.dismoitoutsms.util;

import android.app.Activity;
import android.content.Context;
import android.view.Window;
import android.view.WindowManager;

public class WakeLockManager {
	public static void setWakeUp(Activity context) {
		context.getWindow().addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD
				| WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
				| WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
				| WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
	}
	public static void unsetWakeUp(Activity context) {
		context.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD
				| WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
				| WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
				| WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
	}
}

File: app/src/main/java/fr/piotr/dismoitoutsms/util/WakeLockManager.kt
package fr.piotr.dismoitoutsms.util

import android.view.WindowManager

fun androidx.fragment.app.FragmentActivity.setWakeUp() {

    window.addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD
            or WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
            or WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
            or WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)

}

fun androidx.fragment.app.DialogFragment.setWakeUp() {
    dialog.window?.addFlags(WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD
            or WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
            or WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
            or WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
}